### PR TITLE
DTSPO-7031 Add feature flag to generate an id if a user assigned managed identity is added

### PR DESCRIPTION
**Pull request by zc-hmcts**
### JIRA link ###
https://tools.hmcts.net/jira/browse/DTSPO-7031

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: 12-kubernetes-cluster.tf [hunk 1]
```diff
@@ -58,10 +65,15 @@ resource "azurerm_kubernetes_cluster" "kubernetes_cluster" {
...
 
   identity {
-    type        = "UserAssigned"
+    type         = "UserAssigned"
     identity_ids = [data.azurerm_user_assigned_identity.aks.id]
   }
 
+  kubelet_identity {
+    client_id                 = data.azurerm_user_assigned_identity.kubelet_uami[*].client_id
+    user_assigned_identity_id = data.azurerm_user_assigned_identity.kubelet_uami[*].object_id
```
Comment (timja) [inline 12-kubernetes-cluster.tf:74]: what happens here if you run it with `kubelet_uami_enabled=false`?

Comment (zc-hmcts) [reply 12-kubernetes-cluster.tf:74]: `data.azurerm_user_assigned_identity.kubelet_uami is empty tuple`

File: 12-kubernetes-cluster.tf [hunk 2]
```diff
@@ -58,10 +65,15 @@ resource "azurerm_kubernetes_cluster" "kubernetes_cluster" {
   )
 
   identity {
-    type        = "UserAssigned"
+    type         = "UserAssigned"
     identity_ids = [data.azurerm_user_assigned_identity.aks.id]
   }
 
+  kubelet_identity {
+    client_id                 = var.kubelet_uami_enabled ? data.azurerm_user_assigned_identity.kubelet_uami[0].client_id : []
```
Comment (timja) [inline 12-kubernetes-cluster.tf:73]: does [] work? I would have expected it needed to be `null`

**Review by timja — APPROVED** [review]
nice

**Pull request merged**